src/components/AccountForm/AccountDetails/EditAccountEmail/EditAccountEmail.js

```diff
@@ -0,0 +1,123 @@
+import React, {  useRef } from 'react';
+import { Link, useHistory } from 'react-router-dom';
+import './EditAccountEmail.css';
+
+import { useDispatch, useSelector } from 'react-redux';
+import store from '../../../../store';
+import { emailEdit } from '../../../../reducers/UserSlice';
+import { setErrorMessage } from '../../../../reducers/NotificationSlice';
+
+
+function EditAccountEmail() {
+  const dispatch = useDispatch();
+  const signedIn = useSelector(state => (state.user.userId != null));
+
+  const history = useHistory();
+
+  const oldEmailInput = useRef();
+  const newEmailInput = useRef();
+  const confirmNewEmailInput = useRef();
+
+
+  /**
+   * @function emailEditSubmit - Submit the Edit Email form
+   */
+  
+  async function emailEditSubmit() {
+    // check if the user enters their current email correctly
+    if(store.getState().user.email !== oldEmailInput.current.value) {
+      dispatch(setErrorMessage('Incorrect email!'));
+      return;
+    }
+
+    if (newEmailInput.current.value !== confirmNewEmailInput.current.value) {
+      dispatch(setErrorMessage('Emails do not match!'));
+      return;
+    }
```

Comment:
The way you do these checks are fine, but just so you know that I prefer to group all "verification checks" into a single `try/catch`. I think the only good example of this in the project is in `Room.js` in `roomJoinSubmit`.

This way, the `dispatch(setErrorMessage())` and `return` don't need to be copy/pasted for each `if`. However there is still some copy/pasting even with the `try/catch` way I do,  so it is debatable if one way is even better than the other.